quantized_mesh_tile/editable_terrain.py

```diff
@@ -0,0 +1,487 @@
+# -*- coding: utf-8 -*-
+import numpy
+import os
+
+import numpy as np
+from future.utils import old_div
+
+from quantized_mesh_tile import TerrainTile
+from quantized_mesh_tile.llh_ecef import LLH2ECEF
+from quantized_mesh_tile.terrain import MAX, lerp
+from quantized_mesh_tile.utils import triangleArea
+from . import cartesian3d as c3d
+
+null_normal = [0, 0, 0]
+np_null_normal = np.array([0, 0, 0])
+MIN = 0
+
+
+class EditableTerrainTile(TerrainTile):
+    """
+    Helper class to edit a given terrain tile on the edges.
+    Changes are possible on specified Heights, Triangles and Normals.
+
+
+    """
+
+    def __init__(self, *args, **kwargs):
+        super(EditableTerrainTile, self).__init__(*args, **kwargs)
+        self.is_index_dirty = False
+        self._changed_heights = []
+        self._file_path = None
+        self._gzipped = False
+        self._triangles = []
+        self.name = None
+
+    def set_name(self, name):
+        self.name = name
+
+    def get_edge_vertices(self, edge):
+        """
+        Returns the indices of vertices on the defined edge.
+        :param edge: the edge of the tile. Specified by the flags 'w','n','e','s'
+                    for west-, north-, east- and  south-edge
+        :return: Array of integers
+        """
+        if 'w' == edge:
+            edge_value = MIN
+            search_array = self.u
+        elif 'e' == edge:
+            edge_value = MAX
+            search_array = self.u
+        elif 'n' == edge:
+            edge_value = MAX
+            search_array = self.v
+        elif 's' == edge:
+            edge_value = MIN
+            search_array = self.v
+        indices = [i for i, x in enumerate(search_array) if x == edge_value]
+
+        if len(indices) == 0:
+            raise Exception("No edge vertices found for edge: {}".format(edge))
+        return indices
+
+    def get_edge_coordinates(self, edge):
+        # type: (string) -> Array
+        """
+        Returns a array of coordinate-tupels for all vertices on the specified edge
+        :return: Array of 3d coordinates (3-tupel of float; x,y,z)
+        :param edge: the edge of the tile. Specified by the flags 'w','n','e','s'
+                    for west-, north-, east- and  south-edge
+        """
+        edge_coordinates = []
+        coordinates = self.getVerticesCoordinates()
+        vertices = self.get_edge_vertices(edge)
+        for vertex in vertices:
+            edge_coordinates.append(coordinates[vertex])
+
+        return edge_coordinates
+
+    @property
+    def get_bounding_box(self):
```

Comment:
properties shouldn't start with a `get`

Reply:
This should be maybe put on the parent class

Reply:
OK